### PR TITLE
Better errors for invalid declarations, add more test coverage

### PR DESCRIPTION
**Pull request by Ekrekr**

### CODE REVIEW

File: core/compilers.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function compile(code: string, path: string): string {
   if (path.endsWith(".yaml")) {
     try {
       const yamlAsJson = JSON.stringify(loadYaml(code));
-      return `exports.asJson = () => (${yamlAsJson})`;
+      return `exports.asJson = ${yamlAsJson}`;
```
Comment (BenBirt) [inline core/compilers.ts:26]: why not just `exports = ${yamlAsJson}`?

File: core/compilers.ts [hunk 2]
```diff
@@ -33,8 +33,7 @@ export function compile(code: string, path: string): string {
   }
   if (path.endsWith(".ipynb")) {
     const notebookAsJson = stripNotebookOutputs(JSON.parse(code), path);
-    // TODO(ekrekr): base64 encode the notebook as a string instead.
-    return `exports.asBase64String = () => \`${JSON.stringify(notebookAsJson)}\``;
+    return `exports.asBase64String = \`${JSON.stringify(notebookAsJson)}\``;
```
Comment (BenBirt) [inline core/compilers.ts:36]: this is not a base64 string

File: core/compilers.ts (hunk 1 above)
Comment (Ekrekr) [reply core/compilers.ts:26]: When setting `exports = ...` directly, the return value of `require()` always ends up being `{}`.

The wrapper looks like:

```
(function (exports, require, module, __filename, __dirname) { "use strict"; <return code here> })
```

File: core/compilers.ts (hunk 2 above)
Comment (Ekrekr) [reply core/compilers.ts]: Done.

File: core/compilers.ts [hunk 3]
```diff
@@ -33,8 +33,7 @@ export function compile(code: string, path: string): string {
   }
   if (path.endsWith(".ipynb")) {
     const notebookAsJson = stripNotebookOutputs(JSON.parse(code), path);
```
Comment (BenBirt) [inline core/compilers.ts:35]: I realise this is unrelated to this PR but: I think this is the wrong place for this to be happening! IMO output stripping should be happening in the `compile()` method of a NotebookAction class - following the same pattern for resolving/compiling SQLX/JS code into SQL - not here.

File: core/compilers.ts [hunk 4]
```diff
@@ -33,8 +33,7 @@ export function compile(code: string, path: string): string {
...
-    // TODO(ekrekr): base64 encode the notebook as a string instead.
-    return `exports.asBase64String = () => \`${JSON.stringify(notebookAsJson)}\``;
+    return `exports.asString = \`${JSON.stringify(notebookAsJson)}\``;
```
Comment (BenBirt) [inline core/compilers.ts:36]: IMO this should be returned as JSON, I think that would be a lot cleaner. But I won't block.

**Review by BenBirt — APPROVED** [review]

File: core/compilers.ts (hunk 3 above)
Comment (Ekrekr) [reply core/compilers.ts]: Done.

File: core/compilers.ts (hunk 4 above)
Comment (Ekrekr) [reply core/compilers.ts]: Fair point, it's not really a string! Done.

File: core/compilers.ts (hunk 1 above)
Comment (BenBirt) [reply core/compilers.ts]: It occurred to me: the right way to do this would be:

module.exports = fileAsJson

**Pull request merged**